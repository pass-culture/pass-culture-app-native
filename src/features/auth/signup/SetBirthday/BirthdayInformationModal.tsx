import { t } from '@lingui/macro'
import React from 'react'
import styled from 'styled-components/native'

import { SettingsResponse } from 'api/gen'
import { useDepositAmountsByAge } from 'features/auth/api'
import { AppInformationModal } from 'ui/components/modals/AppInformationModal'
import { BirthdayCake } from 'ui/svg/icons/BirthdayCake'
import { getSpacing, Spacer, Typo } from 'ui/theme'

type Props = {
  settings: SettingsResponse | undefined
  visible: boolean
  hideModal: () => void
}

export const BirthdayInformationModal: React.FC<Props> = ({ settings, visible, hideModal }) => {
  const { eighteenYearsOldDeposit, fifteenYearsOldDeposit } = useDepositAmountsByAge()

  const displayPostGeneralisationMessage =
    settings?.enableNativeEacIndividual && settings.enableUnderageGeneralisation

  const LINE_BREAK = '\n' + '\n'
  const financialHelpMessage = displayPostGeneralisationMessage
    ? t({
        id: 'postGeneralisationFinancialHelpMessage',
        values: {
          deposit15: fifteenYearsOldDeposit.replace(' ', '\u00a0'),
          deposit18: eighteenYearsOldDeposit.replace(' ', '\u00a0'),
        },
        message:
          'Entre 15 et 18 ans, tu es éligible à une aide financière progressive allant de {deposit15} à\u00a0{deposit18}\u00a0offerte par le Gouvernement.',
      }) + LINE_BREAK
    : t({
        id: 'preGeneralisationFinancialHelpMessage',
        values: { deposit: eighteenYearsOldDeposit.replace(' ', '\u00a0') },
        message:
          'Si tu as 18 ans, tu es éligible à une aide financière de\u00a0{deposit}\u00a0offerte par le Gouvernement.',
      }) + LINE_BREAK

  const birthdayInformation =
    t`Nous avons besoin de connaître ton âge.` +
    ' ' +
    financialHelpMessage +
    t`Cette aide sera créditée directement sur ton compte pass Culture.`

  return (
    <AppInformationModal
      title={t`Pourquoi saisir ma date de naissance\u00a0?`}
      numberOfLinesTitle={3}
      visible={visible}
      onCloseIconPress={hideModal}
      testIdSuffix="birthday-information">
      <ModalChildrenContainer>
        <BirthdayCake />
        <Spacer.Column numberOfSpaces={2} />
        <StyledBody>{birthdayInformation}</StyledBody>
      </ModalChildrenContainer>
    </AppInformationModal>
  )
}

const StyledBody = styled(Typo.Body)({
  textAlign: 'center',
})

const ModalChildrenContainer = styled.View({
  paddingTop: getSpacing(5),
  alignItems: 'center',
})
