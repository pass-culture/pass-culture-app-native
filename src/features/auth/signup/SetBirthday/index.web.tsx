import loadable from '@loadable/component'
import React from 'react'

import { LoadingPage } from 'ui/components/LoadingPage'

export const SetBirthday = loadable(() => import('./SetBirthday'), {
  resolveComponent: (components) => components.SetBirthday,
  fallback: <LoadingPage />,
})
