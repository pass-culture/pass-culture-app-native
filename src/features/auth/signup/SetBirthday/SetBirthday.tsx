import { t } from '@lingui/macro'
import React, { FunctionComponent, useState } from 'react'
import styled, { useTheme } from 'styled-components/native'

import { BirthdayInformationModal } from 'features/auth/signup/SetBirthday/BirthdayInformationModal/BirthdayInformationModal'
import { DatePickerDropDown } from 'features/auth/signup/SetBirthday/DatePicker/DatePickerDropDown.web'
import { DatePickerSpinner } from 'features/auth/signup/SetBirthday/DatePicker/DatePickerSpinner'
import { MINIMUM_DATE, UNDER_YOUNGEST_AGE } from 'features/auth/signup/SetBirthday/utils/constants'
import { useDatePickerErrorHandler } from 'features/auth/signup/SetBirthday/utils/useDatePickerErrorHandler'
import { PreValidationSignupStepProps } from 'features/auth/signup/types'
import { analytics } from 'libs/analytics'
import { formatDateToISOStringWithoutTime } from 'libs/parsers'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { ButtonTertiary } from 'ui/components/buttons/ButtonTertiary'
import { useModal } from 'ui/components/modals/useModal'
import { InfoPlain } from 'ui/svg/icons/InfoPlain'
import { Spacer } from 'ui/theme'
import { Form } from 'ui/web/form/Form'

export const SetBirthday: FunctionComponent<PreValidationSignupStepProps> = (props) => {
  const DEFAULT_SELECTED_DATE = new Date(
    new Date().setFullYear(new Date().getFullYear() - UNDER_YOUNGEST_AGE)
  )
  const MAXIMUM_SPINNER_DATE = new Date(DEFAULT_SELECTED_DATE.getFullYear(), 11, 31)

  const { isTouch } = useTheme()
  const { visible, showModal: showInformationModal, hideModal } = useModal(false)
  const [date, setDate] = useState<Date | undefined>()

  const birthdate = date ? formatDateToISOStringWithoutTime(date) : undefined
  const { isDisabled, errorMessage } = useDatePickerErrorHandler(date)

  function onPressWhy() {
    analytics.logConsultWhyAnniversary()
    showInformationModal()
  }

  function goToNextStep() {
    if (birthdate) {
      props.goToNextStep({ birthdate })
    }
  }

  const DatePicker = isTouch ? DatePickerSpinner : DatePickerDropDown

  return (
    <Form.MaxWidth>
      <InnerContainer>
        <ButtonTertiary
          icon={InfoPlain}
          wording={t`Pour quelle raison\u00a0?`}
          accessibilityLabel={t`Pourquoi me demande-t-on ma date de naissance\u00a0?`}
          onPress={onPressWhy}
        />
        <Spacer.Column numberOfSpaces={2} />
        <DatePicker
          onChange={setDate}
          errorMessage={errorMessage}
          defaultSelectedDate={DEFAULT_SELECTED_DATE}
          maximumDate={MAXIMUM_SPINNER_DATE}
          minimumDate={MINIMUM_DATE}
        />
        <Spacer.Column numberOfSpaces={2} />
        <ButtonPrimary
          testID="date-picker-submit-button"
          wording={t`Continuer`}
          accessibilityLabel={props.accessibilityLabelForNextStep}
          disabled={isDisabled}
          onPress={goToNextStep}
        />
        <Spacer.Column numberOfSpaces={5} />
      </InnerContainer>
      <BirthdayInformationModal visible={visible} hideModal={hideModal} />
    </Form.MaxWidth>
  )
}

const InnerContainer = styled.View({
  width: '100%',
  alignItems: 'center',
})
