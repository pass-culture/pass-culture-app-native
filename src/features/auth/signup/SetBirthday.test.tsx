import { StackScreenProps } from '@react-navigation/stack'
import { fireEvent, render, RenderAPI } from '@testing-library/react-native'
import mockdate from 'mockdate'
import React from 'react'
import { Linking } from 'react-native'
import waitForExpect from 'wait-for-expect'

import { goBack, navigate } from '__mocks__/@react-navigation/native'
import { RootStackParamList } from 'features/navigation/RootNavigator'
import { analytics } from 'libs/analytics'
import { ColorsEnum } from 'ui/theme'

import { SetBirthday } from './SetBirthday'

describe('SetBirthday Page', () => {
  beforeEach(() => {
    mockdate.set(new Date('2020-12-01T00:00:00Z'))
    jest.useFakeTimers()
    jest.clearAllMocks()
  })

  it('should render properly', () => {
    const { toJSON } = renderSetBirthday()
    expect(toJSON()).toMatchSnapshot()
  })

  it('should navigate to the previous page on back navigation', () => {
    const { getByTestId } = renderSetBirthday()
    const leftIcon = getByTestId('leftIcon')
    fireEvent.press(leftIcon)

    expect(goBack).toBeCalledTimes(1)
  })

  it('should keep disabled the button "Continuer" when the date is not complete', () => {
    const renderAPI = renderSetBirthday()

    changeDate(renderAPI, '1', '1', '1')

    const button = renderAPI.getByTestId('button-container-validate-birthday')
    expect(button.props.style.backgroundColor).toEqual(ColorsEnum.PRIMARY_DISABLED)
  })

  it('should display the error message "date incorrecte" when the date is too old', () => {
    const renderAPI = renderSetBirthday()

    changeDate(renderAPI, '31', '12', '1889')

    const message = renderAPI.queryByText('La date choisie est incorrecte')
    expect(message).toBeTruthy()
  })

  it('should display the error message "date incorrecte" when the date is incorrect', () => {
    const renderAPI = renderSetBirthday()

    changeDate(renderAPI, '?!', '01', '2002')

    const message = renderAPI.queryByText('La date choisie est incorrecte')
    expect(message).toBeTruthy()
  })

  it('should display the error message "tu dois avoir 16 ans" when the date is too young', () => {
    const renderAPI = renderSetBirthday()

    changeDate(renderAPI, '01', '01', '2005')

    const message = renderAPI.queryByText("Tu dois avoir 16 ans pour t'inscrire")
    expect(message).toBeTruthy()
  })

  it('should enable the button "Continuer" when the date is correct', () => {
    const renderAPI = renderSetBirthday()

    changeDate(renderAPI, '16', '01', '1995')

    const continueButton = renderAPI.getByText('Continuer')
    fireEvent.press(continueButton)

    const buttonContainer = renderAPI.getByTestId('button-container-validate-birthday')
    expect(buttonContainer.props.style.backgroundColor).toEqual(ColorsEnum.PRIMARY)
    expect(renderAPI.queryByText('La date choisie est incorrecte')).toBeFalsy()
    expect(navigate).toBeCalledWith('AcceptCgu', {
      birthday: '1995-01-16',
      email: 'john.doe@example.com',
      isNewsletterChecked: true,
      password: 'password',
    })
  })

  it('should display a information modal when clicking "Pourquoi" link', () => {
    const { getByTestId } = renderSetBirthday()

    const whyBirthdayLink = getByTestId('button-title-why-link')
    fireEvent.press(whyBirthdayLink)

    const birthdayModal = getByTestId('modal-birthday-information')
    expect(birthdayModal.props.visible).toBeTruthy()
  })

  it('should redirect to the "reCAPTCHA privacy" page', async () => {
    jest.spyOn(Linking, 'canOpenURL').mockResolvedValue(true)

    const { getByTestId } = renderSetBirthday()

    const link = getByTestId('external-link-google-data-privacy')
    fireEvent.press(link)

    await waitForExpect(() => {
      expect(Linking.openURL).toHaveBeenCalledWith('https://policies.google.com/privacy')
    })
  })

  it('should redirect to the "reCAPTCHA terms" page', async () => {
    jest.spyOn(Linking, 'canOpenURL').mockResolvedValue(true)

    const { getByTestId } = renderSetBirthday()

    const link = getByTestId('external-link-google-cgu')
    fireEvent.press(link)

    await waitForExpect(() => {
      expect(Linking.openURL).toHaveBeenCalledWith('https://policies.google.com/terms')
    })
  })

  it('should display 4 step dots with the third one as current step', () => {
    const { getAllByTestId } = renderSetBirthday()
    const dots = getAllByTestId('dot-icon')
    expect(dots.length).toBe(4)
    expect(dots[2].props.fill).toEqual(ColorsEnum.PRIMARY)
  })

  it('should call clear focuses', () => {
    const { getByTestId, getByText } = renderSetBirthday()
    const quitSignupModalButton = getByTestId('rightIconButton')

    fireEvent.press(quitSignupModalButton)

    expect(getByText("Continuer l'inscription")).toBeTruthy()
  })

  describe('SetBirthday - analytics', () => {
    it('should log WhyAnniversary when clicking "Pourquoi" link', () => {
      const { getByTestId } = renderSetBirthday()

      const whyBirthdayLink = getByTestId('button-title-why-link')
      fireEvent.press(whyBirthdayLink)

      expect(analytics.logClickWhyAnniversary).toHaveBeenCalledTimes(1)
    })

    it('should log CancelSignup when clicking on "Abandonner l\'inscription"', () => {
      const { getByTestId, getByText } = renderSetBirthday()

      const rightIcon = getByTestId('rightIcon')
      fireEvent.press(rightIcon)

      const abandonButton = getByText("Abandonner l'inscription")
      fireEvent.press(abandonButton)

      expect(analytics.logCancelSignup).toHaveBeenCalledTimes(1)
      expect(analytics.logCancelSignup).toHaveBeenCalledWith('Birthday')
    })

    it('should not log any birthday analytics if the user is 16 years old or more', () => {
      const renderAPI = renderSetBirthday()

      changeDate(renderAPI, '01', '12', '2004')
      expect(analytics.logSignUpBetween14And15Included).not.toBeCalled()
      expect(analytics.logSignUpLessThanOrEqualTo13).not.toBeCalled()

      changeDate(renderAPI, '01', '12', '2000')
      expect(analytics.logSignUpBetween14And15Included).not.toBeCalled()
      expect(analytics.logSignUpLessThanOrEqualTo13).not.toBeCalled()
    })

    it('should log between 14 and 15 birthday analytics if the user is 15 years old', () => {
      const renderAPI = renderSetBirthday()

      changeDate(renderAPI, '01', '12', '2005')

      expect(analytics.logSignUpBetween14And15Included).toBeCalledTimes(1)
      expect(analytics.logSignUpLessThanOrEqualTo13).not.toBeCalled()
    })

    it('should log between 14 and 15 birthday analytics if the user is 14 years old', () => {
      const renderAPI = renderSetBirthday()

      changeDate(renderAPI, '01', '12', '2006')

      expect(analytics.logSignUpBetween14And15Included).toBeCalledTimes(1)
      expect(analytics.logSignUpLessThanOrEqualTo13).not.toBeCalled()
    })

    it('should log <= 13 birthday analytics if the user is 13 years old', () => {
      const renderAPI = renderSetBirthday()

      changeDate(renderAPI, '01', '12', '2007')

      expect(analytics.logSignUpBetween14And15Included).not.toBeCalled()
      expect(analytics.logSignUpLessThanOrEqualTo13).toBeCalledTimes(1)
    })

    it('should log <= 13 birthday analytics if the user is less than 13 years old', () => {
      const renderAPI = renderSetBirthday()

      changeDate(renderAPI, '01', '12', '2010')

      expect(analytics.logSignUpBetween14And15Included).not.toBeCalled()
      expect(analytics.logSignUpLessThanOrEqualTo13).toBeCalledTimes(1)
    })

    it('it should not should log between 14 and 15 birthday analytics more than once', () => {
      const renderAPI = renderSetBirthday()

      changeDate(renderAPI, '01', '12', '2005')
      expect(analytics.logSignUpBetween14And15Included).toBeCalledTimes(1)

      changeDate(renderAPI, '01', '12', '2006')
      expect(analytics.logSignUpBetween14And15Included).toBeCalledTimes(1)
    })

    it('it should not should log <= 13 birthday analytics more than once', () => {
      const renderAPI = renderSetBirthday()

      changeDate(renderAPI, '01', '12', '2007')
      expect(analytics.logSignUpLessThanOrEqualTo13).toBeCalledTimes(1)

      changeDate(renderAPI, '01', '12', '2010')
      expect(analytics.logSignUpLessThanOrEqualTo13).toBeCalledTimes(1)
    })
  })
})

function renderSetBirthday() {
  const navigationProps = {
    route: {
      params: { email: 'john.doe@example.com', isNewsletterChecked: true, password: 'password' },
    },
  } as StackScreenProps<RootStackParamList, 'SetBirthday'>
  return render(<SetBirthday {...navigationProps} />)
}

function changeDate(renderAPI: RenderAPI, dayStr: string, monthStr: string, yearStr: string) {
  const day = renderAPI.getByPlaceholderText('JJ')
  const month = renderAPI.getByPlaceholderText('MM')
  const year = renderAPI.getByPlaceholderText('AAAA')

  fireEvent.changeText(day, dayStr)
  fireEvent.changeText(month, monthStr)
  fireEvent.changeText(year, yearStr)
}
