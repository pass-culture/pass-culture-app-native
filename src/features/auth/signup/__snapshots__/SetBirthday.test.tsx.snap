// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetBirthday Page should render properly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <RNSVGSvgView
      align="none"
      bbHeight="100%"
      bbWidth="100%"
      focusable={false}
      height="100%"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      vbHeight={667}
      vbWidth={375}
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1376171,
                1,
                -13500266,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="prefix__b"
            x1="65.805%"
            x2="47.731%"
            y1="26.588%"
            y2="116.28%"
          />
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1376171,
                1,
                -13500266,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="prefix__d"
            x1="29.678%"
            x2="32.571%"
            y1="71.761%"
            y2="14.036%"
          />
          <RNSVGPath
            d="M0 0h375v667H0z"
            name="prefix__a"
          />
          <RNSVGPath
            d="M574.848 685.018c44.64 237.462 197.943-58.73 172.765-186.83-25.178-128.1-120.317-395.218-387.196-385.031-266.88 10.187-281.873 242.282-174.83 360.46 107.043 118.179 344.621-26.06 389.26 211.401z"
            name="prefix__e"
          />
        </RNSVGDefs>
        <RNSVGGroup
          fill={null}
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        >
          <RNSVGMask
            fill={4294967295}
            height="100%"
            maskContentUnits={1}
            maskTransform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            maskUnits={0}
            name="prefix__c"
            propList={
              Array [
                "fill",
              ]
            }
            width="100%"
            x="0%"
            y="0%"
          >
            <RNSVGUse
              height={0}
              href="prefix__a"
              width={0}
              x={0}
              y={0}
            />
          </RNSVGMask>
          <RNSVGUse
            fill={
              Array [
                1,
                "prefix__b",
              ]
            }
            height={0}
            href="prefix__a"
            matrix={
              Array [
                -1,
                0,
                0,
                1,
                375,
                0,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            width={0}
            x={0}
            y={0}
          />
          <RNSVGGroup
            mask="prefix__c"
          >
            <RNSVGUse
              fill={
                Array [
                  1,
                  "prefix__d",
                ]
              }
              height={0}
              href="prefix__e"
              matrix={
                Array [
                  -0.8660254037844387,
                  -0.49999999999999994,
                  0.49999999999999994,
                  -0.8660254037844387,
                  247.47746608921344,
                  990.3613871835053,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              width={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>,
  <View
    onPress={[Function]}
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <View
      keyboardHeight={0}
      style={
        Array [
          Object {
            "bottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderTopLeftRadius": 16,
              "borderTopRightRadius": 16,
              "bottom": 0,
              "flexDirection": "column",
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
              "position": "absolute",
              "shadowColor": "#151515",
              "shadowOffset": Object {
                "height": 8,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 4,
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-start",
                "flexBasis": 0,
                "flexGrow": 0.2,
                "flexShrink": 1,
                "marginLeft": -4,
                "opacity": 1,
              }
            }
            testID="leftIconButton"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={32}
              bbWidth={32}
              focusable={false}
              height={32}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 32,
                    "width": 32,
                  },
                ]
              }
              testID="leftIcon"
              vbHeight={32}
              vbWidth={32}
              width={32}
            >
              <RNSVGGroup>
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={4279571733}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGPath
                          d="M8.94 19.466c-.301.277-.778.266-1.065-.025-.287-.29-.276-.75.026-1.027l5.314-4.878c1.746-1.603 4.49-1.603 6.237 0l5.314 4.878c.302.277.313.737.026 1.027-.287.29-.764.302-1.065.025l-5.315-4.878c-1.164-1.069-2.993-1.069-4.158 0L8.94 19.466z"
                          matrix={
                            Array [
                              -6.123233995736766e-17,
                              1,
                              1,
                              6.123233995736766e-17,
                              -0.3329999999999984,
                              -0.3329999999999984,
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <Text
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 0.6,
                  "flexShrink": 1,
                  "textAlign": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "#151515",
                    "fontFamily": "Montserrat-Medium",
                    "fontSize": 18,
                    "lineHeight": 22,
                  },
                ]
              }
            >
              Ton anniversaire
            </Text>
          </Text>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "flexBasis": 0,
                "flexGrow": 0.2,
                "flexShrink": 1,
                "marginRight": -4,
                "opacity": 1,
              }
            }
            testID="rightIconButton"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={32}
              bbWidth={32}
              focusable={false}
              height={32}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 32,
                    "width": 32,
                  },
                ]
              }
              testID="rightIcon"
              vbHeight={32}
              vbWidth={32}
              width={32}
            >
              <RNSVGGroup>
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={4279571733}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGPath
                          d="M23.3 8.225c.302-.29.804-.301 1.12-.025.318.276.33.735.028 1.024l-5.322 5.11c-1.166 1.12-1.166 2.879 0 3.998l5.322 5.11c.302.29.29.749-.027 1.025-.317.276-.819.265-1.12-.025l-5.323-5.11c-1.749-1.68-1.749-4.318 0-5.997zm-14.6 0l5.322 5.11c1.749 1.679 1.749 4.318 0 5.997L8.7 24.442c-.302.29-.804.3-1.12.025-.318-.276-.33-.735-.028-1.025l5.322-5.11c1.166-1.12 1.166-2.878 0-3.998l-5.322-5.11c-.302-.29-.29-.748.027-1.024s.819-.265 1.12.025z"
                          matrix={
                            Array [
                              -1,
                              0,
                              0,
                              1,
                              32,
                              0,
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "display": "flex",
                "flexDirection": "column",
                "maxWidth": 500,
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "marginVertical": 40,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "marginBottom": 4,
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": 4,
                    },
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  keyboardType="number-pad"
                  maxLength={2}
                  numberOfChar={2}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="JJ"
                  placeholderTextColor="#626262"
                  rejectResponderTermination={true}
                  selectTextOnFocus={true}
                  style={
                    Array [
                      Object {
                        "color": "#151515",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 15,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "fontSize": 18,
                        "lineHeight": 22,
                        "minWidth": 36,
                        "paddingHorizontal": 10,
                        "textAlign": "center",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <View
                  numberOfSpaces={1}
                  style={
                    Array [
                      Object {
                        "height": 4,
                      },
                    ]
                  }
                />
                <View
                  backgroundColor="#c7c7cc"
                  isEmpty={true}
                  isFocused={false}
                  isValid={false}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#c7c7cc",
                        "borderRadius": 22,
                        "height": 5,
                        "width": "100%",
                      },
                    ]
                  }
                  testID="datepart-bar-day"
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "marginBottom": 4,
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": 4,
                    },
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  keyboardType="number-pad"
                  maxLength={2}
                  numberOfChar={2}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  placeholder="MM"
                  placeholderTextColor="#626262"
                  rejectResponderTermination={true}
                  selectTextOnFocus={true}
                  style={
                    Array [
                      Object {
                        "color": "#151515",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 15,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "fontSize": 18,
                        "lineHeight": 22,
                        "minWidth": 36,
                        "paddingHorizontal": 10,
                        "textAlign": "center",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <View
                  numberOfSpaces={1}
                  style={
                    Array [
                      Object {
                        "height": 4,
                      },
                    ]
                  }
                />
                <View
                  backgroundColor="#c7c7cc"
                  isEmpty={true}
                  isFocused={false}
                  isValid={false}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#c7c7cc",
                        "borderRadius": 22,
                        "height": 5,
                        "width": "100%",
                      },
                    ]
                  }
                  testID="datepart-bar-month"
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "marginBottom": 4,
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": 4,
                    },
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  keyboardType="number-pad"
                  maxLength={4}
                  numberOfChar={4}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  placeholder="AAAA"
                  placeholderTextColor="#626262"
                  rejectResponderTermination={true}
                  selectTextOnFocus={true}
                  style={
                    Array [
                      Object {
                        "color": "#151515",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "Montserrat-Regular",
                        "fontSize": 15,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "fontSize": 18,
                        "lineHeight": 22,
                        "minWidth": 72,
                        "paddingHorizontal": 10,
                        "textAlign": "center",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <View
                  numberOfSpaces={1}
                  style={
                    Array [
                      Object {
                        "height": 4,
                      },
                    ]
                  }
                />
                <View
                  backgroundColor="#c7c7cc"
                  isEmpty={true}
                  isFocused={false}
                  isValid={false}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#c7c7cc",
                        "borderRadius": 22,
                        "height": 5,
                        "width": "100%",
                      },
                    ]
                  }
                  testID="datepart-bar-year"
                />
              </View>
            </View>
          </View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ff99be",
                "borderRadius": 24,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "maxWidth": 500,
                "opacity": 1,
                "paddingBottom": 2,
                "paddingLeft": 2,
                "paddingRight": 2,
                "paddingTop": 2,
                "width": "100%",
              }
            }
            testID="button-container"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 15,
                    "lineHeight": 20,
                    "marginLeft": 5,
                    "maxWidth": 650,
                  },
                ]
              }
              testID="button-title"
              textColor="#ffffff"
            >
              Continuer
            </Text>
          </View>
        </View>
        <View
          customHeight={8}
          style={
            Array [
              Object {
                "height": 8,
              },
            ]
          }
        />
      </View>
    </View>
  </View>,
]
`;
